### PR TITLE
[SYCL][UR] Disable UR_LAYER_PARAMETER_VALIDATION by default

### PR DESCRIPTION
**Pull request by nrspruit**

### CODE REVIEW

File: sycl/source/detail/ur.cpp
Comment (kbenzie) [inline sycl/source/detail/ur.cpp:116]: This is certainly a way of doing this. Ideally validation would be enabled in CI, though, to stop invalid API usage issues slipping through.

Comment (nrspruit) [reply sycl/source/detail/ur.cpp:116]: Do you have a different method to enable this in CI but not in normal runtime?

Comment (kbenzie) [reply sycl/source/detail/ur.cpp:116]: Path of least resistance, based on this approach, would be to update the GitHub Actions workflows to set `UR_LOG_VALIDATION` during testing.

My personal preference would be to have this enabled by default and disabled by a CMake flag but that has it's own drawbacks in that release process would need to be updated to disabled validation by default.

Comment (kbenzie) [reply sycl/source/detail/ur.cpp:116]: The validation layer can also be enabled via `UR_ENABLE_LAYERS=UR_LAYER_PARAMETER_VALIDATION` which is separate from `UR_LOG_VALIDATION` which will output additional info.

File: sycl/test-e2e/format.py
Comment (aelovikov-intel) [inline sycl/test-e2e/format.py:206]: Do we really need it? We had internal e2e tests running with UR verification/L0 plugin leaks detection enabled (before PI removal) and the only single issue caught by UR verification level was the verification code itself.

File: sycl/test-e2e/format.py
Comment (kbenzie) [inline sycl/test-e2e/format.py:205]: I don't think this should be limited to only being enabled on `level_zero:gpu`.

File: sycl/test-e2e/format.py
Comment (kbenzie) [reply sycl/test-e2e/format.py:206]: The plugins used to have some level of parameter checking in the entry points themselves, that's not the case with UR where this only occurs on the validation layer. The failing tests in this PR appear to depend on parameter checking for correctness.

File: sycl/test-e2e/format.py
Comment (kbenzie) [reply sycl/test-e2e/format.py:205]: Also, I'm unsure if this the correct place to set environment variables for an e2e run.

**Review by kbenzie — APPROVED** [review]

**Comment by nrspruit** [comment]
@intel/llvm-gatekeepers please merge when available.

**Comment by nrspruit** [comment]
@intel/llvm-gatekeepers , please merge.

**Comment by sarnex** [comment]
@nrspruit We still need a review from @intel/llvm-reviewers-runtime 

@aelovikov-intel provided initial feedback, maybe he can check if his feedback was addressed and approve if so?

File: sycl/test-e2e/format.py
Comment (aelovikov-intel) [reply sycl/test-e2e/format.py]: 1) I'm still not convinced.
2) > The failing tests in this PR appear to depend on parameter checking for correctness.

can you elaborate on this?

Comment (nrspruit) [reply sycl/test-e2e/format.py]: Hello @aelovikov-intel ,

So, I want to understand your concern. My primary concern is to disable the validation layer by default because this is impacting performance on every call to the UR. Is your concern the fact that we would enable the validation layer during E2E testing?

Why would that be an issue? Don't we want to have more checks during the E2E testing?

Comment (aelovikov-intel) [reply sycl/test-e2e/format.py]: > Why would that be an issue? 

because it affects timing and could possible hide issues that our customers would see without that validation. Also, how many issues that actually catches?

File: sycl/source/detail/ur.cpp
Comment (sarnex) [inline sycl/source/detail/ur.cpp:113]: probably this comment needs to be updated too

**Comment by nrspruit** [comment]
This requires the fix in https://github.com/oneapi-src/unified-runtime/pull/2011 where the opencl adapter must do the alignment check always not only during debug.

**Review by aelovikov-intel — APPROVED** [review]

**Pull request merged**
